src/cmd/linuxkit/run_vcenter.go

```diff
@@ -0,0 +1,292 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type vmConfig struct {
+	vCenterURL  *string
+	dsName      *string
+	networkName *string
+	vSphereHost *string
+
+	vmName     *string
+	path       *string
+	persistent *int64
+	vCpus      *int
+	mem        *int64
+	poweron    *bool
+}
+
+func runVcenter(args []string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var newVM vmConfig
+
+	flags := flag.NewFlagSet("vCenter", flag.ExitOnError)
+	invoked := filepath.Base(os.Args[0])
+
+	newVM.vCenterURL = flags.String("url", os.Getenv("VCURL"), "URL in the format of https://username:password@host/sdk")
+	newVM.dsName = flags.String("datastore", os.Getenv("VMDATASTORE"), "The Name of the DataStore to host the VM")
+	newVM.networkName = flags.String("network", os.Getenv("VMNETWORK"), "The VMware vSwitch the VM will use")
+	newVM.vSphereHost = flags.String("hostname", os.Getenv("VMHOST"), "The Server that will run the VM")
+
+	newVM.vmName = flags.String("vmname", "", "Specify a name for virtual Machine")
+	newVM.path = flags.String("path", "", "Path to a specific image")
+	newVM.persistent = flags.Int64("persistentSize", 0, "Size in MB of persistent storage to allocate to the VM")
+	newVM.mem = flags.Int64("mem", 1024, "Size in MB of memory to allocate to the VM")
+	newVM.vCpus = flags.Int("cpus", 1, "Amount of vCPUs to allocate to the VM")
+	newVM.poweron = flags.Bool("powerOn", false, "Power On the new VM once it has been created")
+
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s push vcenter [options] [name]\n\n", invoked)
+		fmt.Printf("'name' specifies the full path of an image file which will be uploaded\n")
+		fmt.Printf("Options:\n\n")
+		flags.PrintDefaults()
+	}
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatalln("Unable to parse args")
+	}
+
+	remArgs := flags.Args()
+	if len(remArgs) == 0 {
+		fmt.Printf("Please specify the prefix to the image to push\n")
+		flags.Usage()
+		os.Exit(1)
+	}
+	prefix := remArgs[0]
```

Comment:
Please make this be the path, and extract a prefix if you need one. See the changes in qemu yesterday. We are phasing out use of prefix other than for hyperkit kernel_initrd format